crates/sui-framework/sources/tx_context.move

```diff
@@ -51,8 +49,10 @@ module sui::tx_context {
        self.epoch_timestamp_ms
     }
 
-    /// Generate a new, globally unique object ID with version 0
-    public(friend) fun new_object(ctx: &mut TxContext): address {
+    /// Create an `address` that has not been used. As it is an object address, it should never
+    /// occurr as the address for a user.
+    /// In other words, the generated address is a globally unique object ID.
```

Comment:
```suggestion
    /// Create an `address` that has not been used. As it is an object address, it will never
    /// occur as the address for a user.
    /// In other words, the generated address is a globally unique object ID.
```

Just tweaking the wording here -- `should` makes it seem like we're hoping it's true, but I'm guessing we want to make that a guarantee?

Reply:
It is a guarantee in that you would need to be able to find a hash collision. I'm not sure how to phrase that concisely